src/core/Tree/TreeNode.tsx

```diff
@@ -88,24 +68,34 @@ export const TreeNode = (props: TreeNodeProps) => {
     [depthLevel, style],
   );
 
-  const [expanded, setExpanded] = React.useState(isExpanded);
-  const [active, setActive] = React.useState(isActive);
+  const context = React.useContext(TreeContext);
+  const [expanded, setExpanded] = React.useState(false);
+  const isSelected = React.useMemo(() => {
+    return context?.selectedNode === label;
+  }, [context?.selectedNode, label]);
 
   return (
     <li role='treeitem' aria-expanded={expanded} {...rest}>
       <div
         className={cx('iui-tree-node', {
-          'iui-active': active,
+          'iui-active': isSelected,
           'iui-disabled': isDisabled,
           className,
         })}
         style={style_level}
-        onClick={() => {
-          setActive(!active);
-          onSelected?.();
+        onClick={(e) => {
+          if (e.currentTarget != e.target) {
+            return;
+          }
+          if (isSelected) {
+            context?.setSelectedNode('');
+          } else if (!isDisabled) {
+            context?.setSelectedNode(label);
+            context?.onNodeSelected?.();
```

Comment:
I think we need to pass something to `onNodeSelect` because people might want to know what node was selected e.g. to highlight element in a viewer. And I start to think that `onNodeSelected` should go into `TreeNode` as a prop or we need to have some id prop like Material UI:
On tree:
![image](https://user-images.githubusercontent.com/36186912/145772836-2476c9c8-9d07-4273-b732-d4a5e29fb117.png)
On node:
![image](https://user-images.githubusercontent.com/36186912/145772890-59f0ed66-f386-4080-8c03-52bbbbd6e526.png)

Reply:
@mayank99 @veekeys Would be great to get your input on this one too.

Reply:
Could just move it to the individual nodes.

Reply:
Yeah, depends.. If we still going to end up having ids (or some other identifiers) over nodes, having a callback on Tree component might make sense.

Reply:
it does not make sense now empty callbacks on `Tree` component... they give no info at all to the user, so whats the point of having them like that? or its not finished implementation?

Reply:
Yes, it is not finished implementation yet. 
I added `Array<string>` to `onNodeSelected` callback so now you can have a list of the currently selected nodeIds, this allows to have single/multi selection of nodes. If we like this approach I can do something similar for checkboxes/expansion to have list of which nodes are checked/expanded.